addons/account_analytic_default/account_analytic_default.py

```diff
@@ -72,83 +54,76 @@ def account_get(self, cr, uid, product_id=None, partner_id=None, user_id=None, d
         return res
 
 
-class account_invoice_line(osv.osv):
+class account_invoice_line(models.Model):
     _inherit = "account.invoice.line"
     _description = "Invoice Line"
 
-    def product_id_change(self, cr, uid, ids, product, uom_id, qty=0, name='', type='out_invoice', partner_id=False, fposition_id=False, price_unit=False, currency_id=False, company_id=None, context=None):
-        res_prod = super(account_invoice_line, self).product_id_change(cr, uid, ids, product, uom_id, qty, name, type, partner_id, fposition_id, price_unit, currency_id=currency_id, company_id=company_id, context=context)
-        rec = self.pool.get('account.analytic.default').account_get(cr, uid, product, partner_id, uid, time.strftime('%Y-%m-%d'), company_id=company_id, context=context)
-        if rec:
-            res_prod['value'].update({'account_analytic_id': rec.analytic_id.id})
-        else:
-            res_prod['value'].update({'account_analytic_id': False})
+    @api.multi
+    def product_id_change(self, product, uom_id, qty=0, name='', type='out_invoice', partner_id=False, fposition_id=False, price_unit=False, currency_id=False, company_id=None):
+        res_prod = super(account_invoice_line, self).product_id_change(product, uom_id, qty, name, type, partner_id, fposition_id, price_unit, currency_id=currency_id, company_id=company_id)
+        rec = self.env['account.analytic.default'].account_get(product, partner_id, self.env.uid, fields.Date.context_today(self), company_id)
+        res_prod['value'].update({'account_analytic_id': rec and rec.analytic_id.id})
```

Comment:
Use:
res_prod['value']['account_analytic_id'] = rec and rec.analytic_id.id
to add a single key/value pair.

Instead of:
res_prod['value'].update({'account_analytic_id': rec and rec.analytic_id.id})